tree/dataframe/inc/ROOT/RDF/RVariedAction.hxx

```diff
@@ -76,21 +76,43 @@ class R__CLING_PTRCHECK(off) RVariedAction final : public RActionBase {
    }
 
 public:
-   RVariedAction(std::vector<Helper> &&helpers, const ColumnNames_t &columns, std::shared_ptr<PrevNode> prevNode,
-                 const RColumnRegister &colRegister)
-      : RActionBase(prevNode->GetLoopManagerUnchecked(), columns, colRegister, prevNode->GetVariations()),
-        fHelpers(std::move(helpers)), fPrevNodes(MakePrevFilters(prevNode)), fInputValues(GetNSlots())
+   void SetupClass()
    {
+      // The column register and names are private members of RActionBase
+      const auto &colRegister = GetColRegister();
+      const auto &columnNames = GetColumnNames();
+
       fLoopManager->Register(this);
 
-      for (auto i = 0u; i < columns.size(); ++i) {
-         auto *define = colRegister.GetDefine(columns[i]);
+      for (auto i = 0u; i < columnNames.size(); ++i) {
+         auto *define = colRegister.GetDefine(columnNames[i]);
          fIsDefine[i] = define != nullptr;
          if (fIsDefine[i])
             define->MakeVariations(GetVariations());
       }
    }
 
+   RVariedAction(std::vector<Helper> &&helpers, const ColumnNames_t &columns, std::shared_ptr<PrevNode> prevNode,
+                 const RColumnRegister &colRegister)
+      : RActionBase(prevNode->GetLoopManagerUnchecked(), columns, colRegister, prevNode->GetVariations()),
+        fHelpers(std::move(helpers)),
+        fPrevNodes(MakePrevFilters(prevNode)),
+        fInputValues(GetNSlots())
+   {
+      SetupClass();
+   }
+
+   /// This constructor takes in input a vector of previous filter nodes, motivated by the CloneAction logic.
```

Comment:
consider making the ctor private if it is only meant to be used by `RVariedAction::CloneAction`